### PR TITLE
[CS160-38] added assignment CRUD with authentication

### PR DESCRIPTION
**Pull request by ChrisLouie10**
also added authentication for courses.

### CODE REVIEW

File: models/AssignmentModel.js [hunk 1]
```diff
@@ -0,0 +1,64 @@
+const mongoose = require("mongoose");
+const Schema = mongoose.Schema;
+
+
+const AssignmentSchema = new Schema({
+  assignmentName: {
+    type: String,
+    required: true
+  },
+  AssignmentDescription: {
```
Comment (cesantos647) [inline models/AssignmentModel.js:10]: I don't believe that this should be capitalized

File: validation/courseValidation.js [hunk 2]
```diff
@@ -7,8 +7,5 @@ module.exports = Joi.object({
   courseSection: Joi.string().required().messages({
     'any.required': 'courseSection is required'
   }),
-  professorId: Joi.string().required().messages({
```
Comment (cesantos647) [inline validation/courseValidation.js:10]: How come professorId was removed? It's still needed in the schema

**Review by cesantos647 — COMMENTED** [review]
Not sure how async and await works, will need to ask you about that. Other than that, I didn't see anything wrong. Will still need to test.

File: validation/courseValidation.js (hunk 2 above)
Comment (ChrisLouie10) [reply validation/courseValidation.js:10]: Changed it from requiring professorId in the request body to set the professorId as the id of the user making the request (getting the id from the jwt). This way, you can only create/change/delete courses for yourself.

File: models/AssignmentModel.js (hunk 1 above)
Comment (ChrisLouie10) [reply models/AssignmentModel.js]: fixed

**Pull request merged**